src/engine/pieces/king.ts

```diff
@@ -1,13 +1,81 @@
-import Piece from './piece';
+import Piece, { PieceType } from './piece';
 import Player from '../player';
 import Board from '../board';
+import Square, { SquareState } from '../square';
+import GameSettings from '../gameSettings';
 
 export default class King extends Piece {
     public constructor(player: Player) {
-        super(player);
+        super(player, PieceType.KING);
     }
 
     public getAvailableMoves(board: Board) {
-        return new Array(0);
+        let validMoves = new Array(0);
+
+        let changes = [1, -1, 0];
+        let currentSquare = board.findPiece(this);
+
```

Comment:
maybe a more descriptive name here would be good?

Reply:
Actually looking at this more I was thinking it could be nice to define another directions list here, then we can commonise this with knight and maybe other places into a helper function which takes in a list of directions and maps it to a set of valid moves